src/Components/AllListings.jsx

```diff
@@ -6,18 +6,31 @@ class AllListings extends React.Component {
   constructor(props) {
     super(props);
     this.scroller = React.createRef();
+    this.state = {
+      scrollPosition: 0,
+      maxScroll: Infinity,
+    }
   }
 
   render() {
     const { data } = this.props;
-    if (data) {
+    const { scrollPosition } = this.state;
+    const { maxScroll } = this.state;

```

Comment:
You can combine these two into one line

const { scrollPosition, maxScroll } = this.state;